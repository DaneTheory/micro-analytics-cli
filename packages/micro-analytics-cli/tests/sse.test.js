const { listen, mockDb } = require('./utils');
jest.mock('flat-file-db', () => mockDb);
const db = require('../src/db');
const sseHandler = require('../src/sse');
let url;

beforeAll(() => {
  db.initDbAdapter({ adapter: 'flat-file-db' });
});

beforeEach(() => {
  mockDb._reset();
});

describe('sse', () => {
  it('should publish events', () => {
    const sendMock = jest.fn();
    const eventListenerMock = jest.fn();
    sseHandler({ send: sendMock, on: eventListenerMock });

    db.put('superview', { views: [{ time: 1490432584312 }] });

    expect(sendMock).toHaveBeenCalled();
  });

  it('should publish events with correct structure', () => {
    const sendMock = jest.fn();
    const eventListenerMock = jest.fn();
    sseHandler({ send: sendMock, on: eventListenerMock });

    db.put('superview', { views: [{ time: 1490432584312 }] });

    expect(sendMock.mock.calls).toMatchSnapshot();
  });
});
