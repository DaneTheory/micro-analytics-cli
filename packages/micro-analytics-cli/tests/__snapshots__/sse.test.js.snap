// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sse should publish events with correct structure 1`] = `
Array [
  Array [
    Object {
      "data": "{\\"key\\":\\"superview\\",\\"value\\":{\\"views\\":[{\\"time\\":1490432584312}]}}",
      "event": "micro-analytics-ping",
    },
  ],
]
`;
